tests/suites/test_suite_pem.function

```diff
@@ -51,3 +51,26 @@ exit:
     mbedtls_pem_free( &ctx );
 }
 /* END_CASE */
+
```

Comment:
Rather than duplicate the existing test data and much of the test code, please call `mbedtls_pem_read_buffer_with_len` from `mbedtls_pem_read_buffer` as well, using a copy of the buffer without a null terminator (`ASSERT_ALLOC( buf, strlen( data ) )`). And call `mbedtls_pem_read_buffer_with_len` on `data` with the null terminator as well, to check that this also works. And check that the value returned for `use_len` is the same in all cases.

Reply:
Note that https://github.com/ARMmbed/mbedtls/pull/5508 and https://github.com/ARMmbed/mbedtls/pull/5504 are also touching PEM testing (seems to be a hot topic at the moment!).